api/main.py

```diff
@@ -19,7 +19,6 @@
 
 app = FastAPI()
 db = Database()
-oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
 auth = Authentication(db)
```

Comment:
Maybe we could have:
```python
auth = Authentication(db)
oauth2_scheme = auth.get_scheme('token')
```
to avoid having `"token"` defined in `api.auth`.
Or:
```python
auth = Autnehtication(db, token_url='token')
```
and then `token: str = Depends(auth.scheme)` with internally scheme being `self._oauth2_scheme` and a `@property` to get it.

These are minor tweaks, but I think it's better to address issues related to implementation layers as soon as they appear to avoid creating some spaghetti code later on when the code base starts growing.

Reply:
Yes. Also added `user_scopes` to the constructor.